examples/workflows/tidbcloud_dedicated_tier/main.tf

```diff
@@ -0,0 +1,97 @@
+variable "cluster_name" {
+  type     = string
+  nullable = false
+}
+
+variable "cloud_provider" {
+  type     = string
+  nullable = false
+  default  = "AWS"
+}
+
+variable "cloud_provider_region" {
+  type     = string
+  nullable = false
+  default  = "us-east-1"
+}
+
+variable "password" {
+  type      = string
+  nullable  = false
+  sensitive = true
+}
+
+variable "tidb" {
+  type = object({
+    node_size     = string
+    node_quantity = number
+  })
+  nullable = false
+  default = {
+    node_size     = "8C16G"
+    node_quantity = 2
+  }
+}
+
+variable "tikv" {
+  type = object({
+    node_size        = string
+    node_quantity    = number
+    storage_size_gib = number
+  })
+  nullable = false
+  default = {
+    node_size        = "8C32G"
+    node_quantity    = 3
+    storage_size_gib = 500
+  }
+}
+
+variable "tiflash" {
+  type = object({
+    node_size        = string
+    node_quantity    = number
+    storage_size_gib = number
+  })
+  nullable = false
+  default = {
+    node_size        = "8C64G"
+    node_quantity    = 1
+    storage_size_gib = 500
+  }
```

Comment:
For serverless, we still need to specify the concrete node_size/node_quantity/storage_size_gib?